### PR TITLE
Group Project - Recipe App

### PR DESCRIPTION
**Pull request by Tufahel**
> In this pull request we have implemented the following requirements:

### Project Requirements:
- [ ] Login page and registration page:
  - Should be built with Devise.
- [ ] Food list:
  - Should display a list of food added by the logged-in user as in the wireframe (for teams with only two people - also display the quantity of a given food).
  - Should lead to a form that allows users to add new food.
- [ ] Recipes list:
  - Should display a list of recipes created by the logged-in user as in the wireframe.
  - Should lead to recipe details.
  - If the user is the owner of the recipe, should allow the user to delete it.
- [ ] Public recipe list:
  - Should display a list of all public recipes ordered by the newest as in the wireframe.
  - Should lead to recipe details.
- [ ] Recipe details:
  - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
  - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- [ ] General shopping list view (only for a group that has less than 3 members):
  - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
  - Should count the total food items and total price of the missing food.
- [ ] Make sure there are no N+1 queries happening.
- [ ] Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: app/views/recipes/public.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            <% if recipe.public == false%>
+                <p><%= recipe.decription %></p>
+            <% else %>
+                <p>Total foods items: <%= RecipeFood.where(recipe_id: recipe.id).count %></p>
+                <% price = RecipeFood.where(recipe_id: recipe.id).reduce(0) { |sum, f| sum + (f.quantity*Food.find(f.food_id).price).to_f } %>
+                <p>Total price: $<%= price %></p>
+            <% end %>
+          </div>
+        </div>
+      </div>
```
Comment (Omar-Muhamad) [inline app/views/recipes/public.html.erb:21]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=in%20the%20wireframe.-,Should%20lead%20to%20recipe%20details.,-Recipe%20details%3A) when you click on a recipe item should lead to recipe details. So please fix that.

File: app/views/recipes/_recipe.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          hour
+         <% end %> </p>
+         
+        <p>Cooking time: <%= recipe.cooking_time %> 
+        <% if recipe.cooking_time > 1  %>
+           hours
+           <% else  %>
+          hour
+         <% end %>
+         </p>
```
Comment (Omar-Muhamad) [inline app/views/recipes/_recipe.html.erb:19]: - [ ] Cooking time and preparation time suppose to be in hours not a whole date, So please fix that.
Your App:
image
The given wireframe:
image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Tufahel and @jmoseka,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/recipes/public.html.erb (hunk 1 above)
Comment (Tufahel) [reply app/views/recipes/public.html.erb]: @Omar-Muhamad, Thanks for your review. We have fixed it.

File: app/views/recipes/_recipe.html.erb (hunk 2 above)
Comment (Tufahel) [reply app/views/recipes/_recipe.html.erb:19]: @Omar-Muhamad, Thanks for your review. We have fixed it.

**Review by Lordkaito — APPROVED** [review]
Hi @Tufahel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**